scripts/prepare-open-api-for-sdk/index.ts

```diff
@@ -0,0 +1,215 @@
+        );
+      }
+    }
+    if (Object.keys(path).length > 0) {
+      paths[pathKey] = path;
+    }
+  }
+
+  // Removing not used parameters
+  const parametersNames = JSON.stringify(paths)
+    .match(/"#\/components\/parameters\/.*?"/g)
+    .map((match) => match.replace('"#/components/parameters/', "").slice(0, -1))
+    .sort();
+
+  const parameters = {};
+  for (const parameterName of parametersNames) {
+    if (!openAPIContent.components.parameters?.[parameterName]) {
+      console.log(`not found ${parameterName} in parameters`);
+      continue;
+    }
+    parameters[parameterName] = removeAdditionalProperties(
+      openAPIContent.components.parameters[parameterName],
+      !languageOptions.simplifyAllObjectsThatHaveAdditionalProperties
+    );
+    if (languageOptions.removeRequiredOnNullable) {
+      parameters[parameterName] = removeRequiredOnNullable(
+        parameters[parameterName]
+      );
+    }
+  }
+
+  // Removing not used schemas
+  const schemas = {};
+  const schemasNamesFoundInPaths = JSON.stringify(paths)
+    .match(/"#\/components\/schemas\/.*?"/g)
+    .map((match) => match.replace('"#/components/schemas/', "").slice(0, -1))
+    .sort();
+
+  const usedParameters = JSON.stringify(paths)
+    .match(/"#\/components\/parameters\/.*?"/g)
+    .map((match) =>
+      match.replace('"#/components/parameters/', "").slice(0, -1)
+    );
+  const schemasNamesFoundInPathsParameters: string[] = [];
+  for (const parameter of usedParameters) {
+    if (!openAPIContent.components.parameters?.[parameter]) {
+      continue;
+    }
+    const schemasNamesFoundInParameter = JSON.stringify(
+      openAPIContent.components.parameters[parameter]
+    )
+      .match(/"#\/components\/schemas\/.*?"/g)
+      .map((match) => match.replace('"#/components/schemas/', "").slice(0, -1));
+    schemasNamesFoundInPathsParameters.push(...schemasNamesFoundInParameter);
+  }
+
+  const allSchemasNames = [
+    ...schemasNamesFoundInPaths,
+    ...schemasNamesFoundInPathsParameters,
+  ];
+
+  for (const schemaName of allSchemasNames) {
+    if (!openAPIContent.components.schemas?.[schemaName]) {
+      console.log(`not found ${schemaName} in schemas`);
+      continue;
+    }
+    schemas[schemaName] = removeAdditionalProperties(
+      openAPIContent.components.schemas[schemaName],
+      !languageOptions.simplifyAllObjectsThatHaveAdditionalProperties
+    );
+    if (languageOptions.removeRequiredOnNullable) {
+      schemas[schemaName] = removeRequiredOnNullable(schemas[schemaName]);
+    }
+  }
+
+  // Finding other schemas uses
+  let lastSchemaStringify = "";
+  while (true) {
+    const schemaStringify = JSON.stringify(schemas);
+    if (lastSchemaStringify === schemaStringify) {
+      break;
+    }
+    lastSchemaStringify = schemaStringify;
+    const schemasNames = schemaStringify
+      .match(/"#\/components\/schemas\/.*?"/g)
+      .map((match) => match.replace('"#/components/schemas/', "").slice(0, -1))
+      .sort();
```

Comment:
Jak chcesz zwrócić posortowana tablicę to lepiej użyć `toSorted()`. Bo to wtedy nie mutuje tablicy wyjściowej co moze mieć jakiś sideEffect. Chociaz tutaj chyba nie będzie to miało złych konsekwencji.

Reply:
Widzę, że też w wielu innych miejsach jest tak użyte jak coś.

Reply:
toSorted jest dostępny tylko w node20+ i przeglądarkach.
Faktycznie sort mutuje parametr wejściowy, ale w tym miejscu parametrem wejściowym jest wynik `JSON.stringify(paths)`. Czyli kopia paths doprowadzona do string-a.
Więc wszystko jest safe.
`paths`  się nie modyfikuje.

Reply:
Choć nie wiem czy to w ogóle potrzebne :)

Reply:
^ slice robi kopie tablicy, jeszcze poxniej